programmer.go

```diff
@@ -144,69 +145,99 @@ func spProgramNetwork(portname string, boardname string, filePath string, authda
 
 	// Prepare a form that you will submit to that URL.
 	_url := "http://" + portname + "/data/upload_sketch_silent"
-	var b bytes.Buffer
-	w := multipart.NewWriter(&b)
-	// Add your image file
-	filePath = strings.Trim(filePath, "\n")
-	f, err := os.Open(filePath)
-	if err != nil {
-		log.Println("Error opening file" + filePath + " err: " + err.Error())
-		return err
-	}
-	fw, err := w.CreateFormFile("sketch_hex", filePath)
-	if err != nil {
-		log.Println("Error creating form file")
-		return err
-	}
-	if _, err = io.Copy(fw, f); err != nil {
-		log.Println("Error copying form file")
-		return err
-	}
-	// Add the other fields
-	if fw, err = w.CreateFormField("board"); err != nil {
-		log.Println("Error creating form field")
-		return err
-	}
-	if _, err = fw.Write([]byte(colonToUnderscore(boardname))); err != nil {
-		log.Println("Error writing form field")
-		return err
-	}
-	// Don't forget to close the multipart writer.
-	// If you don't close it, your request will be missing the terminating boundary.
-	w.Close()
+	var cookies []*http.Cookie
+
+	for {
+		var b bytes.Buffer
+		w := multipart.NewWriter(&b)
+		// Add your image file
+		filePath = strings.Trim(filePath, "\n")
+		f, err := os.Open(filePath)
+		if err != nil {
+			log.Println("Error opening file" + filePath + " err: " + err.Error())
+			return err
+		}
+		fw, err := w.CreateFormFile("sketch_hex", filePath)
+		if err != nil {
+			log.Println("Error creating form file")
+			return err
+		}
+		if _, err = io.Copy(fw, f); err != nil {
+			log.Println("Error copying form file")
+			return err
+		}
+		// Add the other fields
+		if fw, err = w.CreateFormField("board"); err != nil {
+			log.Println("Error creating form field")
+			return err
+		}
+		if _, err = fw.Write([]byte(colonToUnderscore(boardname))); err != nil {
+			log.Println("Error writing form field")
+			return err
+		}
+		// Don't forget to close the multipart writer.
+		// If you don't close it, your request will be missing the terminating boundary.
+		w.Close()
 
-	// Now that you have a form, you can submit it to your handler.
-	req, err := http.NewRequest("POST", _url, &b)
-	if err != nil {
-		log.Println("Error creating post request")
-		return err
-	}
-	// Don't forget to set the content type, this will contain the boundary.
-	req.Header.Set("Content-Type", w.FormDataContentType())
-	if authdata.Username != "" {
-		req.SetBasicAuth(authdata.Username, authdata.Password)
-	}
+		// Now that you have a form, you can submit it to your handler.
+		req, err := http.NewRequest("POST", _url, &b)
+		if err != nil {
+			log.Println("Error creating post request")
+			return err
+		}
+		// Don't forget to set the content type, this will contain the boundary.
+		req.Header.Set("Content-Type", w.FormDataContentType())
+		if authdata.Username != "" {
+			req.SetBasicAuth(authdata.Username, authdata.Password)
```

Comment:
Yes, I think it is just how the diff is displayed. Everything is tabbed in one more level because of the loop for redirect retry.